system/jlib/jfile.hpp

```diff
@@ -667,4 +671,56 @@ inline bool canRename(const char * filename) { return queryFileSystemProperties(
 inline bool canSeekWrite(const char * filename) { return queryFileSystemProperties(filename).canSeekWrite; }
 inline bool hasDirectories(const char * filename) { return queryFileSystemProperties(filename).hasDirectories; }
 
+
+enum class FileWatchEvents
+{
+    none          = 0,
+#ifndef _WIN32
+    // mapped directly for convenience, any other impl. will need to map to FileWatchEvents values
+    acessed       = IN_ACCESS,
+    modified      = IN_MODIFY,
+    metaModified  = IN_ATTRIB,
+    closed        = IN_CLOSE_NOWRITE,
+    closedWrite   = IN_CLOSE_WRITE,
+    opened        = IN_OPEN,
+    movedFrom     = IN_MOVED_FROM,
+    movedTo       = IN_MOVED_TO,
+    created       = IN_CREATE,
+    deleted       = IN_DELETE,
+#else
+    acessed       = 0x001,
+    modified      = 0x002,
+    closed        = 0x004,
+    closedWrite   = 0x008,
+    metaModified  = 0x010,
+    opened        = 0x020,
+    movedFrom     = 0x040,
+    movedTo       = 0x080,
+    created       = 0x100,
+    deleted       = 0x200,
+#endif
+    anyChange     = created | movedTo | movedFrom | closedWrite | modified | metaModified
+};
+
+inline FileWatchEvents operator|(FileWatchEvents a, FileWatchEvents b)
```

Comment:
Should this use BITMASK_ENUM ?

Reply:
yes, I thought something like that had been introduced, but couldn't find it..
Have switched to it, also amended it to add operator &